driver-core/src/main/java/com/datastax/driver/core/ArrayBackedRow.java

```diff
@@ -239,6 +242,23 @@ public InetAddress getInet(String name) {
         return getInet(metadata.getFirstIdx(name));
     }
 
+    public Token getToken(int i) {
+        if (tokenFactory == null)
+            throw new DriverInternalError("Token factory not set. This should only happen at initialization time");
+
+        metadata.checkType(i, tokenFactory.getTokenType().getName());
+
+        ByteBuffer value = data.get(i);
+        if (value == null || value.remaining() == 0)
+            return null;
+
+        return tokenFactory.deserialize(value);
+    }
+
+    public Token getToken(String name) {
+        return getToken(metadata.getFirstIdx(name));
```

Comment:
It's definitively infortunate that cassandra is not terribly consistent on this, but I think nonetheless that we should make our best effort to make things work as intuitively as possible. And I do think that if you can do getToken("i") when you had "token(i)", then you will expect setToken("i") to work for "token(i)" as well. And I'm pretty sure people will try using it (without named placeholders) and will get bitten by it if we don't.